### PR TITLE
Dropdown options of widgets are now searchable 

### PR DESCRIPTION
**Pull request by amitmawkin**
Please refer issue https://github.com/capitalone/Hygieia/issues/692

image

Widgets affected
- Build
- Code Analysis
- Feature
- Deploy

### CODE REVIEW

File: UI/src/components/widgets/build/config.html [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+                          <div ng-bind-html="item.name | highlight: $select.search"></div>
+
+
+                          <div ng-if="$index == $select.items.length-1">
+                              <button
+                                      class="btn btn-success"
+                                      style="width: 100%; margin-top: 10px;"
+                                      ng-click="buildConfig.fetch($select, $event, $select.items.length);"
+                                      ng-disabled="!buildConfig.loading">Load more...</button>
+                              </div>-->
```
Comment (tabladrum) [inline UI/src/components/widgets/build/config.html]: remove these commented codes?

File: UI/src/components/widgets/build/config.html [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
         <form-group class="text-center" input="collectorItemId" errors="{required: 'Please select a collector'}">
-            <select name="collectorItemId"
+            <!--<select name="collectorItemId"
                     class="form-control"
                     ng-model="buildConfig.collectorItemId"
                     ng-options="item.name for item in buildConfig.buildJobs | orderBy: 'name' track by item.value"
                     ng-disabled="buildConfig.toolsDropdownDisabled"
                     required>
                 <option value="">{{buildConfig.toolsDropdownPlaceholder}}</option>
-            </select>
```
Comment (tabladrum) [inline UI/src/components/widgets/build/config.html]: remove commented out codes?

File: UI/src/components/widgets/codeanalysis/config.html [hunk 3]
```diff
@@ -9,26 +9,52 @@
...
 
-                <select name="caCollectorItem"
+                <!--<select name="caCollectorItem"
                         class="form-control"
                         ng-model="caWidget.caCollectorItem"
                         ng-options="(item.displayName + ' - ' + item.description) for item in caWidget.caJobs | orderBy:['collector.name','description'] track by item.id"
                         ng-disabled="caWidget.caJobs.length === 0">
                     <option value="" disabled>{{caWidget.caToolsDropdownPlaceholder}}</option>
-                </select>
+                </select>-->
```
Comment (tabladrum) [inline UI/src/components/widgets/codeanalysis/config.html]: remove commented codes?

File: UI/src/components/widgets/build/config.html (hunk 2 above)
Comment (amitmawkin) [inline UI/src/components/widgets/build/config.html]: lets keep it , as I am still working on the pagination piece

File: UI/src/components/widgets/build/config.html [hunk 4]
```diff
@@ -1,14 +1,41 @@
...
+                          <div ng-if="$index == $select.items.length-1">
+                              <button
+                                      class="btn btn-success"
+                                      style="width: 100%; margin-top: 10px;"
+                                      ng-click="buildConfig.fetch($select, $event, $select.items.length);"
+                                      ng-disabled="!buildConfig.loading">Load more...</button>
+                              </div>-->
+                </ui-select-choices>
+                <ui-select-no-choice>
+                    Dang! We couldn't find any jobs...
```
Comment (jdamick) [inline UI/src/components/widgets/build/config.html]: better message..

File: api/src/main/resources/application.properties [hunk 5]
```diff
@@ -2,7 +2,7 @@
 dbname=dashboarddb
 dbusername=dashboarduser
 dbpassword=1qazxSw2
-dbhost=localhost
-dbport=33333
+dbhost=10.94.166.157
```
Comment (jdamick) [inline api/src/main/resources/application.properties]: revert these @amitmawkin

File: UI/src/components/widgets/build/config.js [hunk 6]
```diff
@@ -83,7 +94,53 @@
             //});
         }
 
-        function submitForm(valid) {
+
+        function paginationFetch($select, $event, x) {
+            if (!$event) {
```
Comment (jdamick) [inline UI/src/components/widgets/build/config.js]: remove that first case, it's unnecessary

**Comment by jdamick** [comment]
LGTM - though i wish there were unit tests for these.....

**Comment by amitmawkin** [comment]
LGTM

**Comment by tabladrum** [comment]
LGTM

**Comment by amitmawkin** [comment]
LGTM

**Comment by amitmawkin** [comment]
LGTM

File: UI/src/components/widgets/build/config.html (hunk 4 above)
Comment (amitmawkin) [inline UI/src/components/widgets/build/config.html]: Better Message provided

File: UI/src/components/widgets/build/config.js (hunk 6 above)
Comment (amitmawkin) [inline UI/src/components/widgets/build/config.js]: Working on paginating the dropdowns options.. will be needing this code..

**Pull request merged**